src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java

```diff
@@ -210,15 +209,22 @@ private void checkLayoutRecursive(MemoryLayout layout) {
                     maxUnpaddedLayout = Long.max(maxUnpaddedLayout, member.byteSize());
                 }
             }
-            checkGroupSize(ul, maxUnpaddedLayout);
+            checkGroup(ul, maxUnpaddedLayout);
         } else if (layout instanceof SequenceLayout sl) {
             checkHasNaturalAlignment(layout);
+            if (sl.elementLayout() instanceof PaddingLayout pl) {
+                throw memberException(sl, pl,
+                        "not supported because a sequence of a padding layout is not allowed");
+            }
             checkLayoutRecursive(sl.elementLayout());
         }
     }
 
-    // check for trailing padding
-    private void checkGroupSize(GroupLayout gl, long maxUnpaddedOffset) {
+    // check elements are not all padding layouts and for trailing padding
+    private void checkGroup(GroupLayout gl, long maxUnpaddedOffset) {
+        if (!gl.memberLayouts().isEmpty() && gl.memberLayouts().stream().allMatch(e -> e instanceof PaddingLayout)) {
```

Comment:
You have fixed the javadoc - but the check for `memberLayouts().isEmpty()` is still here?

Reply:
This seems correct? We don't want to throw an exception for empty layouts.